File: src/002/p002.go
package main

import "fmt"

func solve() int {
	a := 1
	b := 2
	sum := 0

	for b < 4000000 {
		if b%2 == 0 {
			sum += b
		}
		a, b = b, a+b
	}
	return sum
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/002/p002.js
function solve() {
  let sum = 0
  let a = 1
  let b = 1

  while (b < 4000000) {
    if (b % 2 === 0) {
      sum += b
    }
    const temp = b
    b = a + b
    a = temp
  }
  return sum
}

const result = solve()
console.log(result)
